server/tests/integration/test_api_group.py

```diff
@@ -11,176 +11,319 @@
+		inst = self.get_basic_instance()
+		inst.put()
+
+		# Place dummy_student in another group
+		invited = self.accounts['dummy_student']
+		self.model(
+			assignment=self.assignment.key,
+			member=[invited.key, self.accounts['dummy_staff'].key]
+		).put()
+
+		self.post_json(
+			'/{}/{}/add_member'.format(self.name, inst.key.id()),
+			data={'email': invited.email[0]})
+
+		self.assertStatusCode(400)
+		self.assertEqual(inst.invited, [])
+
+		self.post_json(
+			'/{}/{}/add_member'.format(self.name, inst.key.id()),
+			data={'email': invited.email[0]})
+
+		self.assertStatusCode(400)
+		self.assertEqual(inst.invited, [])
+
+	def test_index_arguments(self):
+		self.user = self.accounts['dummy_student2']
+		self.partner = self.accounts['dummy_student3']
+		inst = self.get_basic_instance()
+		inst.put()
+
+		self.get_index(assignment=self.assignment.key.id(), member=self.user.key.id())
+		self.assertEqual(self.response_json[0]['id'], inst.key.id())
+
+		self.get_index(member=self.partner.key.id())
+		self.assertEqual(self.response_json[0]['id'], inst.key.id())
+
+	def test_remove_from_two_member_group(self):
+		self.user = self.accounts['dummy_student']
+		inst = self.model(assignment=self.assignment.key,
+											member=[self.user.key, self.accounts['dummy_student2'].key])
+		inst.put()
+
+		self.post_json(
+			'/{}/{}/remove_member'.format(self.name, inst.key.id()),
+			data={'email': self.user.email[0]})
+
+		self.assertStatusCode(200)
+		self.assertEqual(inst.key.get(), None)
+	
+	def test_add_member_permission(self):
+		""" Tests that add_member checks for permissions """
+		self.group = self.group.put().get()
+		with self.assertRaises(PermissionError):
+			self.API().add_member(self.group, self.accounts['dummy_student'], {})
+	
+	def test_add_member_already_invited(self):
+		""" Tests that repeating invite not allowed """
+		self.group = self.group.put().get()
+		with self.assertRaises(BadValueError):
+			email = self.accounts['dummy_student'].key
+			self.API().invite(self.group, self.accounts['dummy_admin'], {
+				'email': email
+			})
+			self.group = self.group.put().get()
+			self.assertIn(email, self.group.invited)
+			self.API().add_member(self.group, self.accounts['dummy_admin'], {
+				'email': email
+			})
+
+	def test_add_member_already_in_group(self):
+		""" Tests that cannot invite existing member """
+		self.group = self.group.put().get()
+		with self.assertRaises(BadValueError):
+			email = self.accounts['dummy_student3'].key
+			self.assertIn(email, self.group.member)
+			self.API().add_member(self.group, self.accounts['dummy_admin'], {
+				'email': email
+			})
+
+	def test_remove_member_permissions(self):
+		""" TEsts that remove_member checks for permissions """
+		self.group = self.group.put().get()
+		with self.assertRaises(PermissionError):
+			self.API().remove_member(self.group, self.accounts['dummy_student'], {})
+
+	# def test_invite_permissions(self):
+	# 	""" Tests that invtie checks for permissions """
```

Comment:
Fixed. The test is uncommented in the newer #565 PR... Turns out someone wrote `return need.exception()` instead of `raise need.exception()`, so I fixed that.